server/__init__.py

```diff
@@ -5,25 +5,36 @@
 
 from .rest_slicer_cli import genRESTEndPointsForSlicerCLIsInDocker
 from .handlers import process_annotations
-
+from .constants import PluginSettings
+from .docker_resource import validateSettings, DockerResource,DockerCache
+from girder.utility.model_importer import ModelImporter
 _template = os.path.join(
     os.path.dirname(__file__),
     'webroot.mako'
 )
 
 
 def load(info):
+    """
+    images = ModelImporter.model('setting').getDefault(
+            constants.PluginSettings.DOCKER_IMAGES)
+    ModelImporter.model('setting').set(PluginSettings.DOCKER_IMAGES, images)
+    """
     girderRoot = info['serverRoot']
     histomicsRoot = Webroot(_template)
     histomicsRoot.updateHtmlVars(girderRoot.vars)
     histomicsRoot.updateHtmlVars({'title': 'HistomicsTK'})
 
     info['serverRoot'].histomicstk = histomicsRoot
     info['serverRoot'].girder = girderRoot
+    resource = DockerResource()
+    info['apiRoot'].HistomicsTK = resource
+
+    dockerCache = DockerCache(docker_resource.getDockerImages())
+    dockerImages = dockerCache.getDockerImg()
 
-    genRESTEndPointsForSlicerCLIsInDocker(
-        info, 'HistomicsTK',
-        'dsarchive/histomicstk:v0.1.3'
-    )
+    genRESTEndPointsForSlicerCLIsInDocker(info, resource, dockerCache)
```

Comment:
I meant the prior version of `genRESTEndPointsForSlicerCLIsInDocker` docker pulls from which registry docker is configured (default: Dockerhub) to use on that machine. docker registry. So maybe it is better to let that function stay.

And put your stuff in another function that works with the DockerCache into a separate function tentatively called `genRESTEndPointsForSlicerCLIsInDockerCache`